components/x-gift-article/src/GiftArticle.jsx

```diff
@@ -59,7 +72,18 @@ const withGiftFormActions = withActions(
 						state.showFreeArticleAlert = false
 						return state
 					}
-					const { url, isShortened } = await api.getShorterUrl(state.urls.nonGift)
+
+					let nonGiftUrl = state.urls.nonGift
```

Comment:
**Change Request:** Here I think we can avoid the mutation of `let`  and ensure the type never changes by instead use a `URL` constructor to produce the final url.

This would also avoid issues in the future should the search params change such as the original `state.urls.nonGift` not including a param

Something like the below

```js
const nonGiftUrl = new URL(state.urls.nonGift);

if (state.includeHighlights) {
	const { highlightsAccessToken } = await highlightsApiClient.shareHighlights(
		initialProps.article.id
	)
	if (highlightsAccessToken) {
		nonGiftUrl = `${nonGiftUrl}&highlights=${highlightsAccessToken}`
	}
}

const { url, isShortened } = await api.getShorterUrl(nonGiftUrl.toString())
```

Reply:
Sorry I made a mistake when writing this comment. Just editing the suggestion now so it makes sense. I forgot to change something that I copy and pasted from your code

Reply:
That should be better now

Reply:
yeah, that looks good, will update the PR , thanks alot!